omegaconf/basecontainer.py

```diff
@@ -124,7 +124,10 @@ def __delitem__(self, key: Any) -> None:
         ...
 
     def __len__(self) -> int:
-        return self.__dict__["_content"].__len__()  # type: ignore
+        if self._is_none() or self._is_missing() or self._is_interpolation():
+            return 0
+        content = self.__dict__["_content"]
+        return len(content)
```

Comment:
Moved it here without the assert, I think having two different asserts is not worth the near duplication.

Reply:
Agreed! => resolving